store/tikv/2pc.go

```diff
@@ -1523,17 +1523,39 @@ func (c *twoPhaseCommitter) tryAmendTxn(ctx context.Context, startInfoSchema Sch
 		c.doingAmend = true
 		defer func() { c.doingAmend = false }()
 		if keysNeedToLock.len() > 0 {
-			pessimisticLockBo := NewBackofferWithVars(ctx, pessimisticLockMaxBackoff, c.txn.vars)
 			lCtx := &kv.LockCtx{
-				ForUpdateTS:  c.forUpdateTS,
-				LockWaitTime: kv.LockNoWait,
+				Killed:        c.lockCtx.Killed,
+				ForUpdateTS:   c.forUpdateTS,
+				LockWaitTime:  c.lockCtx.LockWaitTime,
+				WaitStartTime: time.Now(),
 			}
-			err = c.pessimisticLockMutations(pessimisticLockBo, lCtx, keysNeedToLock)
-			if err != nil {
-				logutil.Logger(ctx).Warn("amend pessimistic lock has failed", zap.Error(err), zap.Uint64("txnStartTS", c.startTS))
-				return false, err
+			tryTimes := uint(0)
+			retryLimit := config.GetGlobalConfig().PessimisticTxn.MaxRetryCount
+			for tryTimes < retryLimit {
+				pessimisticLockBo := NewBackofferWithVars(ctx, pessimisticLockMaxBackoff, c.txn.vars)
+				err = c.pessimisticLockMutations(pessimisticLockBo, lCtx, keysNeedToLock)
```

Comment:
Seems it'll return error like `conn %d, existErr for key:%s should not be nil"` from cache after first failure with `ErrWriteConflict`.